plumbing/format/packfile/decoder.go

```diff
@@ -174,24 +185,54 @@ func (d *Decoder) decodeObjectsWithObjectStorerTx(count int) error {
 
 // DecodeObject reads the next object from the scanner and returns it. This
 // method can be used in replacement of the Decode method, to work in a
-// interative way
+// interactive way. If you created a new decoder instance using NewDecoderForType
+// constructor, if the object decoded is not equals to the specified one, null will
```

Comment:
To do this in a proper way, we should refactor the Decoder to be an Iterator instead of get the elements count from the Scanner and then call to DecodeObject 'count' times. But this was the previous behavior. Could we do this refactor in another PR?